modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_5_0/UpgradeDDMDataProviderInstance.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dynamic.data.mapping.internal.upgrade.v3_5_0;
+
+import com.liferay.dynamic.data.mapping.constants.DDMPortletKeys;
+
+/**
+ * @author Carolina Barbosa
+ */
+public class UpgradeDDMDataProviderInstance
+	extends com.liferay.portal.upgrade.v7_0_0.UpgradeLastPublishDate {
```

Comment:
@carolmariaabb, why did you extend from this class instead of the class `BaseUpgradeLastPublishDate`? The methods `addLastPublishDateColumn` and `updateLastPublishDates` are implemented on the class `BaseUpgradeLastPublishDate`, therefore I understand that you should extend from this class, right?

Reply:
Hi @natocesarrego,
I agree with you. I thought the same, but the class `com.liferay.dynamic.data.mapping.internal.upgrade.v1_0_0.UpgradeLastPublishDate` extends from `com.liferay.portal.upgrade.v7_0_0.UpgradeLastPublishDate` and then I thought it would be better to do the same.

Reply:
@carolmariaabb, is there any other class extending from`BaseUpgradeLastPublishDate`? Please check it and see if we extend just from `BaseUpgradeLastPublishDate` attends our needs, if so I think we should extend from it.